pkg/common/kubernetes_version.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2020-2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"k8s.io/client-go/kubernetes"
+	"strconv"
+	"strings"
+)
+
+func GetKubernetesVersion(clientset kubernetes.Interface) (*VersionInfo, error) {
```

Comment:
I think that we could change the return type from *VersionInfo -> *version.Info. Thiis is the type returned by the ServerVersion() call. However, please keep the checks that the minor and major are actually present in the response.

Reply:
`version.Info` has a lot of other fields whereas at this point we only care about `Major` and `Minor`. Besides those fields are of string type which means every time we want to use `Major` and `Minor`, we need to convert them to integers another time.

Reply:
Fair enough.